Marlin/buzzer.h

```diff
@@ -20,11 +20,67 @@
  *
  */
 
-#ifndef BUZZER_H
-  #define BUZZER_H
+#ifndef __BUZZER_H__
+#define __BUZZER_H__
 
-  #if HAS_BUZZER
-    void buzz(long duration, uint16_t freq);
-  #endif
+#include "Marlin.h"
+#include "circularqueue.h"
 
-#endif //BUZZER_H
+#define TONE_QUEUE_LENGTH 8
+
+struct tone_t {
+  uint16_t duration;
+  uint16_t frequency;
+};
+
+class Buzzer {
```

Comment:
If these classes are going to be singletons, we should probably make all their data and methods `static` for the reduced code size and overhead. The same should apply to the `mesh_bed_leveling` class, etc. It's a minor hassle, because we have to re-declare all the static data members in the class `.cpp` files. But we have a minor "mandate" to reduce code size and try to keep Marlin fitting on smaller (64M) boards. Maybe that just means 64M boards can't use most extra features…?